### PR TITLE
Color Picker: Validate uniqueness of selected colors

### LINKED ISSUE #20359: Color picker doesn't validate for unique color codes
### Which Umbraco version are you using?

16.2.0

### Bug summary

I noticed color picker doesn't validate for unique values (hex codes) as e.g. values for dropdown, checkbox list and radiobutton list.

Image

I believe it did in Umbraco 13, so it wasn't possible to add duplicate hex codes.

### Specifics

_No response_

### Steps to reproduce

Add multiple values on same color code (hex) in configuration of swatches.

### Expected result / actual result

_No response_

### PR DESCRIPTION
**Pull request by Programeerik**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

If there's an existing issue for this PR then this fixes #20359 

### Description

I've added an check on duplicates after the valid color check, so it's only checked when it's an valid color. 
I've added an function normalize to the internal ColorListValidator class so the logic is much better readable.
The normalize function is for removing the # before checking for duplicate values.
When the count of duplicates is higher than 0 it's not possible to save the data.

image

I've also added an unittest which check if the test returns the duplicate value

### CODE REVIEW

File: src/Umbraco.Infrastructure/PropertyEditors/ColorPickerConfigurationEditor.cs [hunk 1]
```diff
@@ -46,17 +53,52 @@ public IEnumerable<ValidationResult> Validate(object? value, string? valueType,
...
                 {
-                    yield return new ValidationResult($"The value {item.Value} is not a valid hex color", new[] { "items" });
+                    yield return new ValidationResult($"The value {item.Value} is not a valid hex color", ["items"]);
+                }
+
+                var normalized = Normalize(item.Value);
+                if (seen.Add(normalized) is false)
+                {
+                    duplicates.Add(normalized);
                 }
```
[the remark this answers is not in the document]
Comment (AndyButland) [reply src/Umbraco.Infrastructure/PropertyEditors/ColorPickerConfigurationEditor.cs:74]: I think this is OK.  We only need to track duplicates, not the original value.

File: src/Umbraco.Infrastructure/PropertyEditors/ColorPickerConfigurationEditor.cs [hunk 2]
```diff
@@ -46,17 +53,52 @@ public IEnumerable<ValidationResult> Validate(object? value, string? valueType,
...
             }
```
[the remark this answers is not in the document]
Comment (AndyButland) [reply src/Umbraco.Infrastructure/PropertyEditors/ColorPickerConfigurationEditor.cs:75]: Have added the `continue` statement.

**Comment by Programeerik** [comment]
@AndyButland  Great point about the continue addition. When you talk about the validation in the backoffice, do you mean so that the user get realtime feedback?

The serverside code check can be merged. Thanks for the additions and help.

**Comment by AndyButland** [comment]
> When you talk about the validation in the backoffice, do you mean so that the user get realtime feedback?

Could be real-time, but more that we better surface the server-side validation when managing data type configurations.

**Comment by bjarnef** [comment]
@AndyButland should we consider some of these changes in `ValueListUniqueValueValidator` (of course in a new PR):
https://github.com/umbraco/Umbraco-CMS/blob/767894b72303c54084f7a21bbea077a3fad31738/src/Umbraco.Infrastructure/PropertyEditors/ValueListUniqueValueValidator.cs#L47-L52

It seems `.Select(item => item)` is redundant - and I think `HashSet` may perform better.

**Comment by AndyButland** [comment]
Looks like it, certainly the `Select` is redundant but can't see it does any harm. As you say, I don't think it would be part of this PR (which I've merged now anyway!),
